### PR TITLE
SPIN-1840: Fix Instance Rebooting

### PR DESCRIPTION
**Pull request by zanthrash**
The WaitForUpInstanceHealthTask & WaitForDownInstanceHealthTasks would never get into a success state when the Aamzon health state (which is always "Unknown") was used in determining health.

This would cause the "Reboot Instance Task" to hang on these tasks.  The reboot of the instance would complete but the task would appear hung, causing end user confusion.

This fix simply allows success of the tasks if the only health state is Unknown.

In the case of multiple health checks:
For "Up" health check it will accept combinations of "Up" and "Unknown" state for success
For "Down" a combination of "Down", "OutOfService", & "Unknown" state for success

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForDownInstanceHealthTask.groovy [hunk 1]
```diff
@@ -28,8 +28,10 @@ class WaitForDownInstanceHealthTask extends AbstractWaitForInstanceHealthChangeT
       return true
     }
 
+    boolean allAreUnknown = healthProviders.every {it.state == 'Unknown'}
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForDownInstanceHealthTask.groovy]: I'm not sure we should be using this flag here unless the 'only consider platform health' option is selected

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForUpInstanceHealthTask.groovy [hunk 2]
```diff
@@ -28,10 +28,18 @@ class WaitForUpInstanceHealthTask extends AbstractWaitForInstanceHealthChangeTas
       return false
     }
 
-    if (healthProviders.any { it.state != "Up"}) {
-      return false
+    if (allUp(healthProviders) || mixOfUpAndUnknown(healthProviders)) {
```
Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForUpInstanceHealthTask.groovy]: the original looks wrong, but here I think we want at least one UP, and the rest up or unknown. and in the case of only platform health then allow that to be considered up

**Comment by cfieber** [comment]
there is a `HealthHelper` class that has some of the platform health logic rolled up in it, and the WaitForUpInstancesTask (plural) and WaitForAllInstancesDownTask are using those

I wonder if we couldn't just kill off these single instance tasks and use the corresponding multi-instances task to reduce the proliferation of this health checking logic?

**Comment by zanthrash** [comment]
@cfieber PTAL when you have a moment

**Comment by cfieber** [comment]
LGTM, thanks!

**Pull request merged**